### PR TITLE
fix(rpc): broken ws upgrade introduced by using http.TimeoutHandler

### PR DESCRIPTION
**Pull request by zale144**
# PR Standards

## Opening a pull request should be able to meet the following requirements

---

Close #XXX

<-- Briefly describe the content of this pull request -->

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

**Comment by zale144** [comment]
> can we use stdlib instead https://pkg.go.dev/net/http#TimeoutHandler ?

Check the title dude 😄

**Comment by mtsitrin** [comment]
@zale144 can u pls elaborate what was the issue and how this change fix it?
looking for timeouts+wss support I can find various resources:
* https://github.com/zeromicro/go-zero/blob/master/rest/handler/timeouthandler.go
* https://github.com/golang/go/issues/54136

**Comment by danwt** [comment]
Also I guess if the old solution failed at runtime we should have a regression test for it

**Comment by zale144** [comment]
@mtsitrin @danwt I got errors trying to connect with tendermint, because `http.Hijacker` was not implemented with `http.TimeoutHandler`. I would suggest not going too far with this, as long as we have timeouts and able to upgrade to ws, I'm happy. I won't lose any sleep if we don't have both at the same time (ws with timeout). wdyt?

**Comment by danwt** [comment]
^ Makes sense
I think WS usually have their own idle connection lifetime anyway

**Comment by danwt** [comment]
Does it work now?

**Review by mtsitrin — APPROVED** [review]

**Review by danwt — DISMISSED** [review]
trusting it works

File: rpc/server.go [hunk 1]
```diff
@@ -200,7 +189,7 @@ func (s *Server) serve(listener net.Listener, handler http.Handler) error {
 	s.Logger.Info("serving HTTP", "listen address", listener.Addr())
 	s.server = http.Server{
 		Handler:           handler,
-		ReadHeaderTimeout: ReadHeaderTimeout,
+		ReadHeaderTimeout: readHeaderTimeout,
 	}
```
Comment (danwt) [inline rpc/server.go:193]: do we want readtimeout too? it's often recommended

Comment (mtsitrin) [reply rpc/server.go:193]: no need IMO, it quite overlaps with the timeouthandler

Comment (omritoptix) [reply rpc/server.go:193]: If I'm not mistaken most rpc methods don't have timeout defined (just the subscribe).

however I guess we could assume that rpc nodes exposed to the public have timeouts defined in the infra level (e.g on the load balancer/reverse-proxy) etc. 

So practically speaking not sure it's a concern. with that being said there is also no arm in adding it I guess.

**Review by omritoptix — DISMISSED** [review]

**Pull request merged**